File: experimental data/RQ3&RQ4/translation_result/success/c/cpw/claude3/CAPR/few-shot-True/cot-ast/candidates-2/check_if_a_number_is_jumbled_or_not/check_if_a_number_is_jumbled_or_not.rs
fn check_if_a_number_is_jumbled_or_not(num: i32) -> i32 {
    let mut num = num;
    if num / 10 == 0 {
        return 1;
    }
    while num != 0 {
        if num / 10 == 0 {
            return 1;
        }
        let digit1 = (num % 10).wrapping_abs();
        let digit2 = ((num / 10) % 10).wrapping_abs();
        if (digit2.wrapping_sub(digit1)).wrapping_abs() > 1 {
            return 0;
        }
        num = num / 10;
    }
    1
}

File: benchmark/c/cpw/check_if_a_number_is_jumbled_or_not/check_if_a_number_is_jumbled_or_not.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int check_if_a_number_is_jumbled_or_not ( int num ) ;
int check_if_a_number_is_jumbled_or_not ( int num ) {
  if ( num / 10 == 0 ) return 1;
  while ( num != 0 ) {
    if ( num / 10 == 0 ) return 1;
    int digit1 = num % 10;
    int digit2 = ( num / 10 ) % 10;
    if ( abs ( digit2 - digit1 ) > 1 ) return 0;
    num = num / 10;
  }
  return 1;
}


